django_mongodb_backend/base.py

```diff
@@ -192,11 +198,27 @@ def _rollback(self):
     def set_autocommit(self, autocommit, force_begin_transaction_with_broken_autocommit=False):
         self.autocommit = autocommit
 
+    def _close(self):
+        pass
+
     @async_unsafe
     def close(self):
-        super().close()
+        # MongoClient is a connection pool and, unlike database drivers that
+        # implement PEP 249, shouldn't be closed by connection.close().
+        pass
```

Comment:
Overriding `close()` to do nothing omits [the call to `validate_thread_sharing()`](https://github.com/django/django/blob/0f5dd0dff3049189a3fe71a62670b746543335d5/django/db/backends/base/base.py#L343-L346) to the superclass implementation. If it's important, we could add that here. There's currently [a test failure](https://github.com/django/django/blob/0f5dd0dff3049189a3fe71a62670b746543335d5/tests/backends/tests.py#L889-L912) without it.

Threading, connection sharing between threads... it's all not much I have experience with, so I'm not too confident about any of this. Perhaps you can help to educate me if any of it is obvious to you, Shane.
https://github.com/django/django/commit/34e248efeca272a2c0f2b15a5347dd66fc0340c9

Reply:
I would skip those threaded tests since pymongo does not implement PEP 249. We intentionally want to always share the same client among all threads.

Reply:
I believe the client is already shared among threads as it's stored in the `DatabaseWrapper._connection_pools` class variable.

The `validate_thread_sharing()` logic ensures that separate instances of `DatabaseWrapper` aren't accessed in separate threads. I'm unsure whether removing this logic has any benefits or drawbacks. Unless you can say definitively, I'd suggest we leave it for now.